rust/hook-worker/src/worker.rs

```diff
@@ -175,26 +186,90 @@ impl<'p> WebhookWorker<'p> {
 
             let client = self.client.clone();
             let retry_policy = self.retry_policy.clone();
+            let kafka_producer = self.kafka_producer.clone();
+            let hog_mode = self.hog_mode;
 
             tokio::spawn(async move {
-                let mut futures = Vec::new();
+                let mut futures = Vec::with_capacity(batch.jobs.len());
+                let mut metadatas = Vec::with_capacity(batch.jobs.len());
 
                 // We have to `take` the Vec of jobs from the batch to avoid a borrow checker
                 // error below when we commit.
                 for job in std::mem::take(&mut batch.jobs) {
                     let client = client.clone();
                     let retry_policy = retry_policy.clone();
 
+                    // TODO: Remove the clone by taking ownership of the metadata
+                    metadatas.push(job.metadata().clone());
+
                     let future =
                         async move { process_webhook_job(client, job, &retry_policy).await };
 
                     futures.push(future);
                 }
 
                 let results = join_all(futures).await;
-                for result in results {
+
+                let mut kafka_ack_futures = Vec::new();
+                for (result, mut metadata) in iter::zip(results, metadatas) {
                     if let Err(e) = result {
                         error!("error processing webhook job: {}", e);
+                    } else if hog_mode {
+                        if let Value::Object(ref mut object) = metadata {
+                            if let Ok(result) = result {
+                                match result {
+                                    WebhookResult::Success(response) => {
+                                        object.insert(
+                                            "asyncFunctionResponse".to_owned(),
+                                            serde_json::Value::String(
+                                                response.status().to_string(),
+                                            ),
+                                        );
+                                    }
+                                    WebhookResult::Failed(error) => {
+                                        object.insert(
+                                            "asyncFunctionResponse".to_owned(),
+                                            serde_json::Value::String(error.to_string()),
+                                        );
+                                    }
+                                    WebhookResult::WillRetry => {
+                                        // Nothing to do
+                                    }
+                                }
+                            }
+                        }
```

Comment:
Not sure if I just suck at rust or not reading it right but the body isn't included right? The goal is to have a payload something like the below:

```
const asyncFunctionResponse = {
    vmResponse: {
        status: fetchResponse.status,
        body: responseBody,
    },
    timings: [{
        kind: 'fetch',
        duration_ms: duration,
    }]
}
```

That way the hog function can use the fetch response body for whatever it might need (we also might add something later to indicate whether the body is needed or not based on whether the fetch request is assigned to a value).

Reply:
Yeah, sorry, that's a few more lines of code I didn't get to yet. Getting the status round-tripped was where I ended yesterday and proves the hard part is working. I'm going to fill in proper `vmResponse` and `timings` soon.